TerminateIllegalWrapper fix

dm-ackerman opened the pull request:
# Description

Fix bug in TerminateIllegalWrapper that set values on the wrapper instead of the env.

Fixes #1176

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

elliottower commented on pettingzoo/utils/wrappers/terminate_illegal.py:
```diff
@@ -42,7 +42,6 @@ def step(self, action: ActionType) -> None:
         if self._prev_obs is None:
             self.observe(self.agent_selection)
         if isinstance(self._prev_obs, dict):
-            assert self._prev_obs is not None
```
Why’d you remove this assertion? Think it should check if it exists first before checking the value, but I guess the assertion will fail either way if not

dm-ackerman replied:
I removed it because it is redundant and will never be called if the value is None.
The line before confirms self._prev_obs is a dict. If it's a dict, it's not None, so the assertion can never fail.

elliottower reviewed (approved):
Overall looks good thanks for making the test too, though it should probably be in the file where the other wrapper tests are done just for simplicity. And have we changed all of the other wrappers to do this self.env.unwrapped business yet?

dm-ackerman commented:
> Overall looks good thanks for making the test too, though it should probably be in the file where the other wrapper tests are done just for simplicity. And have we changed all of the other wrappers to do this self.env.unwrapped business yet?

This was the only one I noticed where unwrapped matters.

I moved the wrapper test in with the others

jjshoots reviewed (approved):
Looks pretty reasonable to me. The test for terminate illegal is a bit hack with the double call to the env rollout, but otherwise looks sane enough.
